infra/wca_on_rails/production/pipeline.tf

```diff
@@ -0,0 +1,313 @@
+    deployment_ready_option {
+      action_on_timeout = "CONTINUE_DEPLOYMENT"
+    }
+
+    terminate_blue_instances_on_deployment_success {
+      action                           = "TERMINATE"
+      termination_wait_time_in_minutes = 5
+    }
+  }
+
+  deployment_style {
+    deployment_type   = "BLUE_GREEN"
+    deployment_option = "WITH_TRAFFIC_CONTROL"
+  }
+
+  auto_rollback_configuration {
+    enabled = true
+    events  = ["DEPLOYMENT_FAILURE"]
+  }
+
+  ecs_service {
+    cluster_name = var.shared.ecs_cluster.name
+    service_name = aws_ecs_service.this.name
+  }
+
+  load_balancer_info {
+    target_group_pair_info {
+      prod_traffic_route {
+        listener_arns = [var.shared.https_listener.arn]
+      }
+
+      target_group {
+        name = var.shared.rails-blue-green[0].name
+      }
+
+      target_group {
+        name = var.shared.rails-blue-green[1].name
+      }
+    }
+  }
+}
+
+resource "aws_s3_bucket" "this" {
+  bucket        = var.name_prefix
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "this" {
+  bucket = aws_s3_bucket.this.id
+
+  rule {
+    id = "rule-1"
+
+    filter {}
+
+    expiration {
+      days = 30
+    }
+
+    status = "Enabled"
+  }
+}
+
+resource "aws_codepipeline" "this" {
+  name     = var.name_prefix
+  role_arn = aws_iam_role.codepipeline_role.arn
+
+  artifact_store {
+    type     = "S3"
+    location = aws_s3_bucket.this.bucket
+  }
+
+  stage {
+    name = "source"
+
+    action {
+      name             = "source"
+      category         = "Source"
+      owner            = "AWS"
+      provider         = "ECR"
+      version          = "1"
+      output_artifacts = ["image"]
+
+      configuration = {
+        RepositoryName = var.shared.ecr_repository.name
+        ImageTag       = "latest"
+      }
+    }
+  }
+
+  stage {
+    name = "build"
+
+    action {
+      name     = "build"
+      category = "Build"
+      owner    = "AWS"
+      provider = "CodeBuild"
+      version  = "1"
+
+      input_artifacts  = ["image"]
+      output_artifacts = ["build_output"]
+
+      configuration = {
+        ProjectName = aws_codebuild_project.build.name
+      }
+    }
+  }
+
+  stage {
+    name = "deploy"
+
+    action {
+      name     = "deploy"
+      category = "Deploy"
+      owner    = "AWS"
+      provider = "CodeDeployToECS"
+      version  = "1"
+
+      input_artifacts = ["build_output"]
+
+      configuration = {
+        ApplicationName                = aws_codedeploy_app.this.name
+        DeploymentGroupName            = aws_codedeploy_deployment_group.this.deployment_group_name
+        TaskDefinitionTemplateArtifact = "build_output"
+        TaskDefinitionTemplatePath     = "taskdef.json"
+        AppSpecTemplateArtifact        = "build_output"
+        AppSpecTemplatePath            = "appspec.yaml"
+        Image1ArtifactName             = "build_output"
+        Image1ContainerName            = "IMAGE_NAME"
+      }
+    }
+  }
```

Comment:
consider fourth notification stage

Reply:
Added notifications on deployment start, success and failure